app/components/SettingsPanel/SettingsPanel.js

```diff
@@ -12,16 +12,39 @@ export default class SettingsPanel extends React.Component {
       newColorB: '',
       newColorC: '',
     };
+    this.stateValidation = {
+      newColorA: { isValid: false, defaultColor: '#f00', },
+      newColorB: { isValid: false, defaultColor: '#00f', },
+      newColorC: { isValid: false, defaultColor: '#0f0', },
+    };
+    this.newColorNames = Object.keys(this.state);
+  }
+  componentWillUpdate(nextProps, nextState) {
+    const { stateValidation, newColorNames, } = this;
+    const isHexColor = colorKey => (
```

Comment:
Apart from https://devdocs.io/html/element/input/color , you should use `input[pattern]` https://devdocs.io/html/attributes#pattern-attribute and use the validity property CSS https://devdocs.io/css/:valid and API https://devdocs.io/dom/validitystate
All of these are documented well here: https://developer.mozilla.org/en-US/docs/Learn/HTML/Forms/Form_validation